plutus-core/plutus-core/src/Crypto.hs

```diff
@@ -17,3 +28,25 @@ verifySignature pubKey msg sig =
             <$> publicKey pubKey
             <*> pure msg
             <*> signature sig
+
+verifySECP256k1Signature
+  :: BS.ByteString -- ^ Public key
+  -> BS.ByteString -- ^ Signature
+  -> BS.ByteString -- ^ Message hash
+  -> Emitter (EvaluationResult Bool)
+verifySECP256k1Signature pk sig msg =
+  case DSIGN.rawDeserialiseVerKeyDSIGN @SECP256k1DSIGN pk of
+    Nothing -> do
+      emitM "SECP256k1: Given invalid signing key."
+      pure EvaluationFailure
+    Just pk' -> case DSIGN.rawDeserialiseSigDSIGN @SECP256k1DSIGN sig of
+      Nothing -> do
+        emitM "SECP256k1: Given invalid signature."
+        pure EvaluationFailure
+      Just sig' -> case SECP.msg msg of
+        Nothing -> do
+          emitM "SECP256k1: Given invalid message hash."
+          pure EvaluationFailure
+        Just msg' -> case DSIGN.verifyDSIGN () pk' msg' sig' of
+          Left _   -> pure . pure $ False
```

Comment:
I don't understand all the stuff in `DSIGN`, but I see that the `_` is a string.  Is that something that would be worth logging in the `Left` case, or is it stuff that should be caught by the earlier checks anyway?